### ISSUE TITLE
Default project creates too long paths

### ISSUE THREAD

**Issue by frankhaugen** [xamarin/Xamarin.Forms#4140]
### Description
When creating a default Xamarin.Forms projact at the default Visual Studio 2017 project location, (C:\users\<username>\repos), it creates paths/filenames that are too long

As an integral part of Visual Studio, Xamarin should't need short project names, as a project might be part of a sub-folder of a repository, which would negate the workaround of placing the project in "C:\repos"

### Steps to Reproduce

1. Start Visual Studio 2017
2. Click "New Project"
3. Choose Visual C#->Cross-Platform->Mobile App (Xamarin.Forms) project type
4. Name it "MyXamarinFormsErrorReplicationProject"
5. Choose to create a directory for the project
6. Choose iOS and Android, then .net standard
7. Without changing anything, choose to debug android project, (I debug on physical device)
8. Project will fail to build/debug

### Expected Behavior
Xamarin/Visual Studio, creating a buildable default project

### Actual Behavior
Building a clean default project fails

#### TL;DR error message:
**System.IO.PathTooLongException: The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.**

#### Detailed error message:
Severity	Code	Description	Project	File	Line	Suppression State
Error		The "GenerateJavaStubs" task failed unexpectedly.
System.IO.PathTooLongException: The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.
   at System.IO.LongPathHelper.Normalize(String path, UInt32 maxPathLength, Boolean checkInvalidCharacters, Boolean expandShortPaths)
   at System.IO.Path.NewNormalizePath(String path, Int32 maxPathLength, Boolean expandShortPaths)
   at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength, Boolean expandShortPaths)
   at System.IO.Path.GetFullPathInternal(String path)
   at System.IO.Path.GetFullPath(String path)
   at Xamarin.Android.Tasks.GenerateJavaStubs.Run(DirectoryAssemblyResolver res)
   at Xamarin.Android.Tasks.GenerateJavaStubs.Execute()
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()	MyXamarinFormsErrorReplicationProject.Android			

### Basic Information

- Version with issue: 3.1.0
- Last known good version: -
- IDE: Visual Studio 2017
- Platform Target Frameworks: 
  - iOS:  Not tested
  - Android: 8.0
- Android Support Library Version: 
- Nuget Packages: Xamarin.Forms 3.1.0.697729
- Affected Devices: Samasung Galaxy S8

**Comment by jonathanpeppers** [comment]
Hi @frankhaugen,

You didn't mention what version of VS 2017 you have, but I suspect 15.8.x?

We made changes to `$(IntermediateOutputPath)` in recent releases, for example:
- 15.7 - `obj\Debug`
- 15.8 - `obj\Debug\MonoAndroid81` (too long!)
- 15.9 - `obj\Debug\81`

Do you have the same problem using latest 15.9?

Also, curious, how long is your username?

**Comment by jonathanpeppers** [comment]
Ok, so the length of the project/solution name causes it: `MyXamarinFormsErrorReplicationProject`

I think by default the template creates subdirectories with that name twice... :man_facepalming:
```
MyXamarinFormsErrorReplicationProject/
MyXamarinFormsErrorReplicationProject/MyXamarinFormsErrorReplicationProject/
MyXamarinFormsErrorReplicationProject/MyXamarinFormsErrorReplicationProject.Android/
```
We could do something in Xamarin.Android here to gain a few more characters, but it is a hard problem. I'm not sure if the new long path support in Windows is usable for us yet, since we depend on Java and Android to use them.

@joj what do you think about changing the template a bit? To get rid of sub-directories? Is that even an idea?

**Comment by joj** [comment]
We've been through this a few times. We don't get much with changing the name from the template (also, the naming is fairly standard for VS). Short names in the projects to begin with help a lot more (like, `FormsError` in your case).

**Comment by jfversluis** [comment]
Easy problem to get around from a user perspective, hard problem to get around technically. I hope this will be better with .NET MAUI!
